src/cmd/support_archive/logs.go

```diff
@@ -0,0 +1,121 @@
+package support_archive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const logCollectorName = "logCollector"
+
+type logCollector struct {
+	collectorCommon
+
+	context   context.Context
+	clientSet kubernetes.Interface
+	namespace string
+}
+
+func newLogCollector(log logr.Logger, supportArchive tarball, context context.Context, clientSet kubernetes.Interface, namespace string) collector { //nolint:revive // argument-limit doesn't apply to constructors
+	return logCollector{
+		collectorCommon: collectorCommon{
+			log:            log,
+			supportArchive: supportArchive,
+		},
+		context:   context,
+		clientSet: clientSet,
+		namespace: namespace,
+	}
+}
+
+func (lc logCollector) Do() error {
+	logInfof(lc.log, "Starting log collection")
+
+	podList, err := lc.getPodList()
+	if err != nil {
+		return err
+	}
+
+	for _, pod := range podList.Items {
+		podGetOptions := metav1.GetOptions{}
+		pod, err := lc.clientSet.CoreV1().Pods(lc.namespace).Get(lc.context, pod.Name, podGetOptions)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		lc.collectPodLogs(pod)
+	}
+	return nil
+}
+
+func (lc logCollector) Name() string {
+	return logCollectorName
+}
+
+func (lc logCollector) getPodList() (*corev1.PodList, error) {
+	listOptions := metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "pod",
+		},
+	}
```

Comment:
maybe overthinking here, BUT what if a customer adds some pod into our namespace?
then we would collect the logs for those containers, which I don't think we want

we should maybe use some label selector or something in our list options to only get what we want 

but also MAYBE if we have some random logs from some random container then it can maybe help us debug

Reply:
if I sound conflicted, that is because I am 😅

Reply:
If in doubt, in this case more is better. Cause if there's an additional pod in our namespace, that's suspicious on its own and hence worth knowing.

Reply:
we should collect the information that other pods are running in our namespace (i.e. pod names), but should not collect the logs of these pods. I don't like collecting logs or random pods that are not owned by us because of privacy reasons.
We should have documentation how to collect these logs when/if we need them, but not collect them by default.

Reply:
Fair enough. Then we keep collecting the existence of enemy pods in our namespace to the next step where we collect K8S objects. So that we at least know something's fishy.

Reply:
I've restricted log collection to pods labeled with `app.kubernetes.io/name=dynatrace-operator`